src/identity/provider/SolidIdentityProvider.ts

```diff
@@ -8,10 +8,11 @@ import type { AnyObject, CanBePromise, Configuration } from 'oidc-provider';
 // eslint-disable-next-line @typescript-eslint/prefer-ts-expect-error, @typescript-eslint/ban-ts-comment
 // @ts-ignore
 import instance from 'oidc-provider/lib/helpers/weak_cache';
+import type { HttpHandlerInput } from '../../server/HttpHandler';
+import type { AsyncHandler } from '../../util/AsyncHandler';
 import { NotFoundHttpError } from '../../util/errors/NotFoundHttpError';
-import { OidcProvider } from './OidcProvider';
 
-export class SolidOidcProvider extends OidcProvider {
```

Comment:
By eleminating "OidcProvider" there is now not a generic term to describe a "Provider" that has a "handle" function. Though I do like the idea of using `Provider implements AsyncHander<HttpHandlerInput>` instead of what I did.

Reply:
When and if we need to create alternative Identity providers, they can easily implement AsyncHandler and if other identity providers were to be implemented, I'm not too sure why they should be extending Panva's Provider.